frontend/src/app/features/work-packages/routing/wp-view-base/work-package-single-view.base.ts

```diff
@@ -191,21 +191,23 @@ export class WorkPackageSingleViewBase extends UntilDestroyedMixin {
     const attachments = this.workPackage.attachments as unknown&{ href:string };
     this.attachmentsResourceService.fetchAttachments(attachments.href).subscribe();
 
-    // Fetch file link collections for work package (only if storages module is enabled)
     if (this.workPackage.$links.fileLinks) {
-      this.fileLinkResourceService.updateCollectionsForWorkPackage(this.workPackage.$links.fileLinks.href as string);
+      this.fileLinkResourceService
+        .updateCollectionsForWorkPackage(this.workPackage.$links.fileLinks.href as string)
+        .subscribe(
+          () => {
+            this.projectsResourceService
+              .lookup((this.workPackage.project as unknown&{ id:string }).id)
+              .pipe(this.untilDestroyed())
+              .subscribe((project) => {
+                if (project._links.storages) {
+                  this.storages.updateCollection(project._links.self.href, project._links.storages);
+                }
+              });
+          },
+        );
```

Comment:
To reduce nesting avoid a new observable in a subscribe, instead use something like this. And do not forget error handling here ;)

```typescript
    if (this.workPackage.$links.fileLinks) {
      this.fileLinkResourceService
        .updateCollectionsForWorkPackage(this.workPackage.$links.fileLinks.href as string)
        .pipe(
          this.untilDestroyed(),
          switchMap(() => this.projectsResourceService.lookup((this.workPackage.project as unknown&{ id:string }).id)),
        )
        .subscribe(
          (project) => {
            if (project._links.storages) {
              this.storages.updateCollection(project._links.self.href, project._links.storages);
            }
          },
          (error) => {
            this.toastService.addError(error);
            throw error;
          },
        );
    }
```

Reply:
Applied, thanks!